docs/proposals/17-ratelimits-for-shoot-dns-entries.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Rate limits for DNS entries created by shoot-dns-service
+gep-number: 17
+creation-date: 2021-05-03
+status: implementable
+authors:
+- "@MartinWeindel"
+reviewers:
+- "@timuthy"
+---
+
+# GEP-17: Rate limits for DNS entries created by shoot-dns-service
+
+## Table of Contents
+
+- [GEP-17: Rate limits for DNS entries created by shoot-dns-service](#gep-17-rate-limits-for-dns-entries-created-by-shoot-dns-service)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [DNS controller manager](#dns-controller-manager)
+    - [Gardener](#gardener)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+This `GEP` introduces rate limits for `DNSEntries` created with the help of the [shoot-dns-service](https://github.com/gardener/gardener-extension-shoot-dns-service) to protect the [dns-controller-manager](https://github.com/gardener/external-dns-management) from being throttled, i.e to protect the DNS management of Gardener from intentional or unintentional *DoS* attacks.
+
+## Motivation
+
+The `dns-controller-manager` runs update requests for DNS record on the configured DNS cloud service(s) like `AWS Route 53`. Typically these DNS cloud services are rate limiting requests on account level. As the `shoot-dns-service` allows to request and update `DNSEntries` directly or indirectly from resources on the shoot cluster, a single shoot cluster can potentially eat up the available request rate of the backend account. The `dns-controller-manager` runs as extension on each seed of a Gardener landscape and they all have to share the account for the DNS cloud service in form of a `DNSProvider` as all shoots use the same default *hosted DNS zone* for `DNSEntries` created by Gardener itself ( `external`, `ingress`) and all others created by the `shoot-dns-service`. It is this default `DNSProvider` which needs to be protected at first place. If the user provides an own `DNSProvider` and uses his hosted DNS zone(s) of his own account, the throttling would only affect this account. But also in this case, the user may want to avoid that one of his shoot clusters can break all his others clusters using the same `DNSProvider`.
+
+### Goals
+
+- The rate limit is set and applied separatedly for each `DNSProvider`
+- All update requests caused by `DNSEntries` created or updated from the `shoot-dns-service` should only be forwarded to the backend DNS service, if the update requests are within rate limit of the `DNSProvider` the `DNSEntry` is assigned to.
```

Comment:
```suggestion
- All update requests caused by `DNSEntries` created or updated from the `shoot-dns-service` should only be processed by the backend DNS service, if the update requests are within rate limit of the `DNSProvider` the `DNSEntry` is assigned to.
```

Because forwarding happens since the `shoot-dns-service` doesn't know anything about rate limits, right?